src/dotnet-install.sh

```diff
@@ -1001,18 +1001,34 @@ downloadcurl() {
     local remote_path_with_credential="${remote_path}${feed_credential}"
     local curl_options="--retry 20 --retry-delay 2 --connect-timeout 15 -sSL -f --create-dirs "
     local failed=false
+    local curl_exit_code=0;
     if [ -z "$out_path" ]; then
-        curl $curl_options "$remote_path_with_credential" 2>&1 || failed=true
+        curl $curl_options "$remote_path_with_credential" 2>&1
+        curl_exit_code=$?
+        if [ $curl_exit_code -gt 0 ]; then
+            failed=true
```

Comment:
We don't seem to need this `failed` variable anymore. It is used only once and we can just use `curl_exit_code -gt 0` instead.